ckb_dlfcn.h

```diff
@@ -259,7 +259,13 @@ int ckb_dlopen2(const uint8_t *dep_cell_hash, uint8_t hash_type,
         if (ph->p_offset < prepad) {
           return ERROR_ELF_NOT_ALIGNED;
         }
-        ret = _ckb_load_cell_code(addr2, memsz, ph->p_offset - prepad, ph->p_filesz,
+        uint64_t offset = ph->p_offset;
+        uint64_t read_size = ph->p_filesz;
+        if (offset % RISCV_PGSIZE != 0) {
```

Comment:
Do we really need to align file offset? I personally wasn't sure if there is a need for that.

Reply:
Actually, this change is identical to :
```
ret = _ckb_load_cell_code(addr2, memsz, ph->p_offset - prepad, ph->p_filesz + prepad, index, CKB_SOURCE_CELL_DEP);
```
The same padding is calculated, twice.

Reply:
In that sense I don't think we need this bulk of changes, we have `prepad` above, which is also used in a condition, we can simply make the one line change, there is no need to calculate the padding again here.

Reply:
`prepad` is calculated from `p_vaddr` so it is not always same as `offset % RISCV_PGSIZE`. It can be adjusted by linker script or compiler. Here, p_vaddr - p_offset % align == 0, align can be 4096, 2048, or even 1.
So new padding value should be calculated again.

Reply:
The `prepad` is calculated from `p_vaddr`. As we know, `p_vaddr` and `p_offset` should satisfy:
`p_vaddr - p_offset % align == 0`, which `align` can be 4096, or 2048, ... even 1, depending or compiler or linker script.
When `align` is not 4096, these paddings are not same.

Reply:
We could debate the formula used to calculate padding, but for one library, there should be only one padding used, so I would suggest either use `prepad` and delete the rest, or keep current implementation, and delete parts related to `prepad`. Current code, IMHO, is slightly messy.

Reply:
Correct: When padding are not same, we should take original prepad: p_vaddr % RISCV_PGSIZE.